engine/src/main/java/org/entando/entando/aps/system/services/digitalexchange/component/DigitalExchangeComponentListProcessor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ * 
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.digitalexchange.component;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.entando.entando.aps.system.services.RequestListProcessor;
+import org.entando.entando.web.common.model.RestListRequest;
+import org.entando.entando.web.digitalexchange.component.DigitalExchangeComponent;
+
+public class DigitalExchangeComponentListProcessor extends RequestListProcessor<DigitalExchangeComponent> {
+
+    private static final String NAME = "name";
+    private static final String LAST_UPDATE = "lastUpdate";
+    private static final String VERSION = "version";
+    private static final String TYPE = "type";
+    private static final String RATING = "rating";
+    private static final String INSTALLED = "installed";
+
+    public DigitalExchangeComponentListProcessor(RestListRequest restListRequest, List<DigitalExchangeComponent> components) {
+        super(restListRequest, components);
+    }
+
+    @Override
+    protected BiFunction<String, String, Predicate<DigitalExchangeComponent>> getPredicates() {
```

Comment:
I liked very much this extraction of predicates and comparators to their own methods that we can reuse